### PR TITLE
Implement keeper for gov xmod

### PR DESCRIPTION
**Pull request by NYBACHOK**

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -13,7 +13,7 @@ members = [
   "gears-derive",
 
   # GAIA
-  "gaia-rs",
+  # "gaia-rs",
```
Comment (joneskm) [inline Cargo.toml]: Uncomment!

Comment (dmvict) [inline Cargo.toml:16]: Why do you do this?

File: x/bank/Cargo.toml [hunk 2]
```diff
@@ -9,7 +9,7 @@ gears = { path = "../../gears", features = ["cli", "export", "xmods"] }
 
 #newtypes
 bytes = { workspace = true }
-ibc-proto = { workspace = true } # TODO: should we import through gears via core-types?
+ibc-proto = { workspace = true }
```
Comment (dmvict) [inline x/bank/Cargo.toml:12]: Why did you remove this?

File: x/gov/src/keeper.rs [hunk 3]
```diff
@@ -133,4 +144,271 @@ impl<SK: StoreKey, PSK: ParamsSubspaceKey, M: Module, BM: Module, BK: BankKeeper
...
+        None => Ok(None),
+    }
+}
+
+fn deposit_set<DB: Database>(
+    mut store: GasKVStoreMut<'_, DB>,
+    deposit: &MsgDeposit,
+) -> Result<(), GasStoreErrors> {
+    store.set(
+        MsgDeposit::key(deposit.proposal_id, &deposit.depositor),
```
Comment (dmvict) [inline x/gov/src/keeper.rs:379]: Can you update signature? It would look like `deposit.generete_key()`.

File: x/gov/src/msg/proposal.rs [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+        Ok(MsgSubmitProposal::decode::<Bytes>(value.value.into())
+            .map_err(|e| CoreError::DecodeProtobuf(e.to_string()))?)
+    }
+}
+
+impl From<MsgSubmitProposal> for Any {
+    fn from(msg: MsgSubmitProposal) -> Self {
+        Any {
+            type_url: MsgSubmitProposal::TYPE_URL.to_string(),
+            value: msg.encode_vec().expect(IBC_ENCODE_UNWRAP),
```
Comment (dmvict) [inline x/gov/src/msg/proposal.rs:116]: If I'm correct, the newer protobuf converts  bytes without result.

File: x/gov/src/msg/vote.rs [hunk 5]
```diff
@@ -0,0 +1,132 @@
...
+#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
+pub struct MsgVote {
+    pub proposal_id: u64,
+    pub voter: AccAddress,
+    pub option: VoteOption,
+}
+
+#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
+pub enum VoteOption {
+    Empty = 0,
```
Comment (dmvict) [inline x/gov/src/msg/vote.rs:26]: Please, use common format for all options.

File: x/gov/src/keeper.rs [hunk 6]
```diff
@@ -133,4 +144,271 @@ impl<SK: StoreKey, PSK: ParamsSubspaceKey, M: Module, BM: Module, BK: BankKeeper
...
+        ctx: &mut TxContext<'_, DB, SK>,
+        MsgSubmitProposal {
+            content,
+            initial_deposit,
+            proposer: _proposer,
+        }: MsgSubmitProposal,
+    ) -> anyhow::Result<u64> {
+        /*
+           in go they perform check is it possible to handle
+           proposal somehow, but not sure we need it and instead
```
Comment (joneskm) [inline x/gov/src/keeper.rs:265]: Perhaps add a link to the cosmos code?

File: x/bank/Cargo.toml (hunk 2 above)
Comment (NYBACHOK) [reply x/bank/Cargo.toml:12]: We had this discussion, or at least I think that we had. We need to export types in gears only if they being used more than once. So this comment is outdated.

Comment (NYBACHOK) [reply x/bank/Cargo.toml:12]: Maybe better example why we should omit re-export of structures is `ibc-rs` xmod which rely on different version of `ibc-proto` then `gears`

File: x/gov/src/msg/proposal.rs (hunk 4 above)
Comment (NYBACHOK) [reply x/gov/src/msg/proposal.rs:116]: I use older from gears here.

File: x/gov/src/msg/vote.rs (hunk 5 above)
Comment (NYBACHOK) [reply x/gov/src/msg/vote.rs:26]: Not sure what you mean?

File: Cargo.toml (hunk 1 above)
Comment (NYBACHOK) [reply Cargo.toml]: Done

File: x/gov/src/keeper.rs (hunk 6 above)
Comment (NYBACHOK) [reply x/gov/src/keeper.rs:265]: Done

File: x/gov/src/msg/vote.rs (hunk 5 above)
Comment (dmvict) [reply x/gov/src/msg/vote.rs:26]: It is better to skip byte representation or add it for each option. 

I know that below contains code with this description. Some formal rules improves reading and it can.

Comment (NYBACHOK) [reply x/gov/src/msg/vote.rs:26]: Added to every option
